ament_cmake_version/cmake/ament_cmake_version.cmake

```diff
@@ -0,0 +1,107 @@
+#
+# /// \def RCUTILS_VERSION_MINOR
+# /// Defines RCUTILS minor version number
+# #define RCUTILS_VERSION_MINOR (@rcutils_VERSION_MINOR@)
+#
+# /// \def RCUTILS_VERSION_PATCH
+# /// Defines RCUTILS version patch number
+# #define RCUTILS_VERSION_PATCH (@rcutils_VERSION_PATCH@)
+#
+# /// \def RCUTILS_VERSION_STR
+# /// Defines RCUTILS version string
+# #define RCUTILS_VERSION_STR "@rcutils_VERSION@"
+#
+# #endif  // RCUTILS__VERSION_H_"
+
+function(ament_cmake_version)
+  include_directories(${CMAKE_CURRENT_BINARY_DIR}/ament_cmake_version/include)
+  set(TMP_INCLUDE_DIR ${CMAKE_CURRENT_BINARY_DIR}/ament_cmake_version/include/${PROJECT_NAME})
+
+  set(${PROJECT_NAME}_VERSION_FILE_NAME ${TMP_INCLUDE_DIR}/version.h)
+  set(NEED_TO_CREATE_VERSION_FILE FALSE)
+
+  # retrieve verson information from <package>.xml file
+  # call ament_package_xml() if it has not been called before
+  if(NOT _AMENT_PACKAGE_NAME)
+    ament_package_xml()
+  endif()
+
+  # parse version information from the version string
+  string(REGEX MATCH "([0-9]+\).([0-9]+)\.([0-9]+)" "" dummy ${${PROJECT_NAME}_VERSION})
+  set(${PROJECT_NAME}_VERSION_MAJOR ${CMAKE_MATCH_1})
+  set(${PROJECT_NAME}_VERSION_MINOR ${CMAKE_MATCH_2})
+  set(${PROJECT_NAME}_VERSION_PATCH ${CMAKE_MATCH_3})
+
+  # sanity checks
+  if("${${PROJECT_NAME}_VERSION_MAJOR}" STREQUAL "")
+    message(FATAL_ERROR "${PROJECT_NAME}_VERSION_MAJOR is empty or non-numeric, check <package>.XML")
+  endif()
+  if("${${PROJECT_NAME}_VERSION_MINOR}" STREQUAL "")
+    message(FATAL_ERROR "${PROJECT_NAME}_VERSION_MINOR is empty or non-numeric, check <package>.XML")
+  endif()
+  if("${${PROJECT_NAME}_VERSION_PATCH}" STREQUAL "")
+    message(FATAL_ERROR "${PROJECT_NAME}_VERSION_PATCH is empty or non-numeric, check <package>.XML")
+  endif()
+
+  # Check if the version file exist
+  if(EXISTS "${${PROJECT_NAME}_VERSION_FILE_NAME}")
+    # The file exists
+    # Check if it contains the same version
+    file(STRINGS ${${PROJECT_NAME}_VERSION_FILE_NAME} VERSION_FILE_STRINGS REGEX "#define ${PROJECT_NAME}_VERSION_STR")
+    if(VERSION_FILE_STRINGS STREQUAL "")
+      set(NEED_TO_CREATE_VERSION_FILE TRUE)
+    else()
+      string(REGEX MATCH "^#define[ \t]+${PROJECT_NAME}_VERSION_STR[ \t]+\"([0-9]+\.[0-9]+\.[0-9]+)\"" "" dummy ${VERSION_FILE_STRINGS})
+      if(NOT (${CMAKE_MATCH_1} STREQUAL ${${PROJECT_NAME}_VERSION}))
+        # Create new file if file version != ${PROJECT_NAME}_VERSION
+        set(NEED_TO_CREATE_VERSION_FILE TRUE)
+      endif()
+    endif()
+  else()
+    # if the version file does not exist, create it
+    set(NEED_TO_CREATE_VERSION_FILE TRUE)
+  endif()
+
+  if(${NEED_TO_CREATE_VERSION_FILE})
+    message(STATUS "Create new version file for version ${${PROJECT_NAME}_VERSION}")
+    file(MAKE_DIRECTORY ${TMP_INCLUDE_DIR})
+    # create the version.h file
+    configure_file("resource/version.h.in" ${${PROJECT_NAME}_VERSION_FILE_NAME} NEWLINE_STYLE UNIX)
```

Comment:
Why should the newline style be enforced to `UNIX` even when running this on Windows?